front/src/reducers/index.ts

```diff
@@ -57,12 +58,26 @@ const saveUserFilter = createFilter('user', userWhiteList);
 
 const saveMainFilter = createFilter('main', mainWhiteList);
 
+const dateTransform = createTransform(
+  null,
+  (outboundState: StdcmPathStep[]) =>
+    outboundState.map((step) => {
+      if (!step.isVia && step.arrival && isISOString(step.arrival as unknown as string)) {
```

Comment:
Why do we need this check? Why do we need this cast?

Are we receiving somehow some values which aren't actually of the type `StdcmPathStep`?

Reply:
It's just a safety check, I can remove it if you want

Reply:
Why is it a string? If it's a string, it's supposed to cause a type error on the side where it's set, no?

Reply:
Turns out we get strings because this function gets data from local storage (where it's serialized as JSON).

Docs: https://github.com/rt2zz/redux-persist?tab=readme-ov-file#transforms

My suggestion: avoid using `StdcmPathStep` since the serialized objects don't satisfy this type.

```ts
// Deserialize date strings coming from local storage
const dateTransform = createTransform(
  null,
  (outboundState: { arrival?: string }[]) =>
    outboundState.map(({ arrival, ...step }) => {
      if (arrival) {
        return { ...step, arrival: new Date(arrival) };
      }
      return step;
    }),
  { whitelist: ['stdcmPathSteps'] }
);
```